src/Me/Navigation/Navbar.tsx

```diff
@@ -20,24 +23,42 @@ const MenuItem = ({
 }: {
   icon: string;
   label: string;
-  to: string;
+  to?: string;
+  onClick?: () => void;
 }) => {
   const router = useRouter();
+  if (to) {
+    return (
+      <Link href={to}>
+        <NavItemDecoration
+          className={classNames({ active: router.pathname === to })}
+        >
+          <Icon />
+          <Label>{label}</Label>
+        </NavItemDecoration>
+      </Link>
+    );
+  }
   return (
-    <Link href={to}>
-      <NavItemDecoration
-        className={classNames({ active: router.pathname === to })}
-      >
+    <button onClick={toggleChat}>
+      <NavItemDecoration>
         <Icon />
         <Label>{label}</Label>
       </NavItemDecoration>
-    </Link>
+    </button>
   );
 };
 
 const Navbar = () => {
   const { isAdmin, logout } = useUser();
   const urls = useRoutes();
+  const { isMobile } = useDeviceType();
+  useEffect(() => {
+    if (!isMobile) return;
+    hideWidget();
+    return () => showWidget();
```

Comment:
Is this cleanup function been called? Isn't this function firing when `isMobile` is changed? If so, what's the case that this variables changes?

Reply:
yes its running everytime ismobile changed. 'whats the case that this variable changes', I dont understand what your trying to say, I did not write the code for isMobile. You wont have the desired behavior if i remove that cleanup. change the width of your browser to mobile size, close the chat, now make the screen wider. The button is gone (its not needed anymore), now how will you open the chat?

Reply:
nevermind I understand what you want me to do.